### PR TITLE
Firebase code update to v4.5.0 with thread safe function calling and …

### PR DESCRIPTION
**Pull request by Javasamurai**
Firebase code update to v4.5.0 with thread safe function calling,bug fixes and other big optimisations from google.

### CODE REVIEW

**Comment by rampr** [comment]
is it tested ?

**Comment by Javasamurai** [comment]
Tested in ipad.No issue.Not sure about the data we would get from firebase.

**Comment by bijoshtj** [comment]
@Javasamurai  please check firebase events and user properties getting tracked.

File: ios/config.json [hunk 1]
```diff
@@ -7,7 +7,10 @@
   "frameworks": [
     "FirebaseAnalytics.framework",
     "FirebaseCore.framework",
+    "FirebaseCoreDiagnostics.framework",
+    "FirebaseNanoPB.framework",
```
Comment (rampr) [inline ios/config.json]: do we need these frameworks ?

File: ios/config.json [hunk 2]
```diff
@@ -7,7 +7,10 @@
...
     "FirebaseInstanceID.framework",
+    "nanopb.framework",
```
Comment (rampr) [inline ios/config.json]: do we need this framework too ?

File: ios/config.json (hunk 1 above)
Comment (Javasamurai) [reply ios/config.json]: Firebasecore depends on FirebaseCoreDiagnostics and FirebaseNanoPB

**Pull request merged**